fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -82,11 +82,14 @@ struct RelocateData {
 };
 
 class ParallelTCInfo : public ReferenceCounted<ParallelTCInfo>, public IDataDistributionTeam {
+private:
+	UID id;
+
 public:
 	vector<Reference<IDataDistributionTeam>> teams;
 	vector<UID> tempServerIDs;
 
-	ParallelTCInfo() { }
+	ParallelTCInfo() : id(deterministicRandom()->randomUniqueID()) {}
```

Comment:
ParallelTCInfo is more like a wrapper to get all teams for the same shard in two DCs.
We should not assign a new randomUID to it. Instead, we should return the `teams`'s IDs.

You can refer to code: `getServerIDs`: https://github.com/xumengpanda/foundationdb/blob/053cbc23690db7af63ee874c7d7ae1ce7cf999ff/fdbserver/DataDistributionQueue.actor.cpp#L140

Reply:
The interface requires the function to return an UID. I will keep it as is.

Reply:
I guess you meant `getTeamID ` must return UID?

Since `getTeamID ()` is added in this PR, I didn't see why `getTeamID()` cannot return `string`? 
In that case, we don't have to assign a randomUID for ParallelTCInfo() because `ParallelTCInfo ` is just a wrapper to get info for teams. 

In `getTeamID ` function below, you can return the concatenation of  the `getTeamID` result from each team of `teams `.

Reply:
Sure the function can return a string, but I preferred UID for the sake of type safety. But I agree that it doesn't make much sense to assign an UID to this type here. Will change the interface to return a string instead.